cmd/gcs/main.go

```diff
@@ -81,6 +86,95 @@ func readMemoryEvents(startTime time.Time, efdFile *os.File, cgName string, thre
 	}
 }
 
+// runWithRestartMonitor starts a command with given args and waits for it to exit. If the
+// command exit code is non-zero the command is restarted with with some back off delay.
+// Any stdout or stderr of the command will be split into lines and written as a log with
+// logrus standard logger.  This function must be called in a separate goroutine.
+func runWithRestartMonitor(arg0 string, args ...string) {
+	backoffSettings := backoff.NewExponentialBackOff()
+	// give after retrying for 10 mins
+	backoffSettings.MaxElapsedTime = time.Minute * 10
+	for {
+		command := exec.Command(arg0, args...)
+		err := command.Run()
```

Comment:
nit: can combine this line with the `if`

Reply:
Fixed.